src/handlers/motions/motionStaked/motionStaked.ts

```diff
@@ -36,10 +39,54 @@ export default async (event: ContractEvent): Promise<void> => {
   if (stakedMotion) {
     const {
       id,
-      motionData: { usersStakes },
+      motionData: { usersStakes, events: updatedEvents },
       motionData,
     } = stakedMotion;
 
+    if (vote.eq(0) && !updatedEvents.find(({ name }) => name === 'ObjectionRaised')) {
+      updatedEvents.push(
+        {
+          name: 'ObjectionRaised',
+          transactionHash,
+          logIndex,
+          initiatorAddress: staker,
+        },
+      );
+    }
+
+    updatedEvents.push(
+      {
+        name: 'MotionStaked',
+        transactionHash,
+        logIndex,
+        initiatorAddress: staker,
+        vote: vote.toString(),
+        amount: amount.toString(),
+      },
+    );
+
+    if (requiredStake.eq(BigNumber.from(motionStakes.raw.yay))) {
+      const eventName = !updatedEvents.find(({ name }) => name === 'ObjectionRaised') ? 'MotionFullyStaked' : 'MotionFullyStakedAfterObjection';
```

Comment:
A faster way to do this than to iterate through all previous messages would be to store a `hasObjection` flag on the MotionData model that you set to true in the event the `vote === MotionVote.Nay`. 

Then:  

``` js

const eventName = hasObjection  ? 'MotionFullyStaked' : 'MotionFullyStakedAfterObjection';

```